File: app/src/main/java/com/example/mohang/mvvmproject/MainActivity.java
package com.example.mohang.mvvmproject;

import android.app.FragmentManager;
import android.app.FragmentTransaction;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

import com.example.mohang.mvvmproject.ui.MvvmFragmment;
import com.example.mohang.mvvmproject.ui.adapters.ActionHandler;

public class MainActivity extends AppCompatActivity implements ActionHandler{

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_mvvm);
        FragmentManager fragmentManager=getFragmentManager();
        FragmentTransaction fragmentTransaction=fragmentManager.beginTransaction();
        fragmentTransaction.add(R.id.container,new MvvmFragmment());
        fragmentTransaction.commit();



    }
}

File: app/src/main/java/com/example/mohang/mvvmproject/MainActivity.kt
package com.example.mohang.mvvmproject

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.util.Log
import com.example.mohang.mvvmproject.ui.MvvmFragmment
import com.example.mohang.mvvmproject.viewmodel.BaseViewModel

class MainActivity : AppCompatActivity() {

    val MY_VIEW_MODEL_TAG = "MainActivity"

    var   myViewModel:BaseViewModel?=null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_mvvm)

        Log.d(MY_VIEW_MODEL_TAG,"oncreate")
        val fragmentManager = fragmentManager
        val fragmentTransaction = fragmentManager.beginTransaction()
        var mvvmFragment = fragmentManager.findFragmentById(R.id.container) ;

        if (mvvmFragment == null) {
            mvvmFragment= MvvmFragmment();
            fragmentTransaction.add(R.id.container, mvvmFragment)
            fragmentTransaction.commit()
        }
    }



    override fun onRestart() {
        super.onRestart()
        Log.d(MY_VIEW_MODEL_TAG,"restart")
    }




}
